pydoctor/model.py

```diff
@@ -712,6 +722,9 @@ def Package(self) -> Type['Package']:
     @property
     def Attribute(self) -> Type['Attribute']:
         return self._factory.Attribute
+    @property
+    def FunctionOverload(self) -> Type['FunctionOverload']:
+        return FunctionOverload
```

Comment:
We can just use plain `FunctionOverload` and remove the property I believe.

Reply:
:+1: I didn't know whether you liked this abstraction (the previous PR references it on this class, but it was impl'd differently at that time). I will just instantiate the overload directly in the AST builder.

Reply:
Yea this factory change is very recent indeed, and all objects on the factory are sharing the same signature. I think it’s a good thing to keep it like that. This is why I don’t think FunctionOverload should be part of it.

Reply:
Removed and instantiating `FunctionOverload` directly.